azure-container-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/containerapps/DeployMojo.java

```diff
@@ -0,0 +1,102 @@
+import com.microsoft.azure.toolkit.lib.common.exception.AzureToolkitRuntimeException;
+import com.microsoft.azure.toolkit.lib.common.operation.AzureOperation;
+import com.microsoft.azure.toolkit.lib.containerapps.config.ContainerAppConfig;
+import com.microsoft.azure.toolkit.lib.containerapps.task.DeployContainerAppTask;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+
+import java.io.IOException;
+
+import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
+
+/**
+ * Deploy your project to target Azure Spring app. If target app doesn't exist, it will be created.
+ */
+@Mojo(name = "deploy")
+@Slf4j
+public class DeployMojo extends AbstractMojoBase {
+
+    private static final int GET_URL_TIMEOUT = 60;
+    private static final int GET_STATUS_TIMEOUT = 180;
+    private static final String PROJECT_SKIP = "Packaging type is pom, taking no actions.";
+    private static final String PROJECT_NO_CONFIGURATION = "Configuration does not exist, taking no actions.";
+    private static final String PROJECT_NOT_SUPPORT = "`azure-container-apps:deploy` does not support maven project with " +
+            "packaging %s, only jar is supported";
+
+    /**
+     * Boolean flag to control whether to wait the deployment status to be ready after deployment
+     */
+    @Parameter(property = "noWait")
+    private Boolean noWait;
+
+    /**
+     * Boolean flag to control whether to prompt the tasks before deployment
+     */
+    @Parameter(property = "prompt")
+    private Boolean prompt;
+
+    @Override
+    @AzureOperation("user/containerapps.deploy_mojo")
+    protected void doExecute() throws Throwable {
+        this.mergeCommandLineConfig();
+        loginAzure();
+        selectSubscription();
+        final ContainerAppConfig config = this.getConfiguration();
+        final DeployContainerAppTask task = new DeployContainerAppTask(config);
+        task.execute();
+    }
+
+    private void mergeCommandLineConfig() {
+        try {
+            final JavaPropsMapper mapper = new JavaPropsMapper();
+            mapper.configure(FAIL_ON_UNKNOWN_PROPERTIES, false);
+            final DeployMojo commandLineConfig = mapper.readSystemPropertiesAs(JavaPropsSchema.emptySchema(), DeployMojo.class);
+            com.microsoft.azure.toolkit.lib.common.utils.Utils.copyProperties(this, commandLineConfig, false);
+        } catch (IOException | IllegalAccessException e) {
+            throw new AzureToolkitRuntimeException("failed to merge command line configuration", e);
+        }
+    }
+
+    protected boolean checkProjectPackaging(MavenProject project) throws MojoExecutionException {
+        if (MavenConfigUtils.isJarPackaging(project)) {
```

Comment:
As discussed offline, we can probably leverage this to further whether the project is spring-boot based. We can start with Jar packaging first.